system/atomic_host.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public licenses
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION='''
+---
+module: atomic_host
+short_description: Manage the atomic host platform
+description:
+    - Manage the atomic host platform
+    - Reboot option for commands ```rollback```, ```upgrade``` and ```deploy``` are not supported, the playbook should do explicit reboot
+version_added: "2.2"
+author: "Saravanan KR @krsacme"
+notes:
+    - Host should be an atomic platform (verified by existence of '/run/ostree-booted' file)
+requirements:
+  - atomic
+options:
+    command:
```

Comment:
So, This playbook looks like it would be great for orchestration, but doesn't really have a good ongoing configuration management story. I would expect there to be a way for me to specify a specific version to be installed (without having to handle the associated logic outside of the module).

For example, I would expect to be able to accomplish the following tasks:

- Ensure a particular version:
```
- atomic_host:
    revision: <some version>
  register: result
- <task to reboot>
   when: result | changed
```

- Or ensuring the latest (though, arguably upgrade does this):
```
- atomic_host:
    revision: latest
  register: result
- <task to reboot>
   when: result | changed
```